src/repositories/request-repository.ts

```diff
@@ -1,14 +1,14 @@
 import { CID } from 'multiformats/cid'
 import type { Connection, EntityManager, InsertResult, UpdateResult } from 'typeorm'
 import TypeORM from 'typeorm'
-const { EntityRepository, Repository, Brackets, LessThan } = TypeORM
+const { EntityRepository, Repository } = TypeORM
+import { DateUtils } from 'typeorm/util/DateUtils.js'
```

Comment:
1. `request-repository.ts` is an ES module
2. ES module imports must use file extensions. Alternatively, for Node.js, one could import an exported name (see below).
3. You could import a file from a package if it is exported, i.e. it is inside `exports` clause in `package.json`. This is similar to `public` vs `private` modifiers in Java or C++. If exported, the file is "public", otherwise it is "private".
4. An exported name could be much different from an actual file name (remember how we tried to add an export to `commons` package some time ago).
5. TypeORM (even our pinned version) uses wild-card `exports` clause: https://github.com/typeorm/typeorm/blob/master/package.json#L335.

All of the typeorm files are exported because of the wild-card export. To import a file from typeorm we have to use `.js` file extension. Otherwise (statement 2), Node.js can not find what file to load.

There are three ultimate sources for this: https://nodejs.org/api/packages.html, https://nodejs.org/api/modules.html and https://nodejs.org/api/esm.html It's rather boring though.

Reply:
geez that's all very complicated.  If there's not a simple rule to follow to know when to do it, is there a simple way we could test whether we did it right?  I feel like some of these issues we've had have been because tests pass locally, but then fail in CI or importing the package fails later.  Is there something we could put into our build process to prevent bad import statements from slipping in?